module/VuFind/src/VuFind/Db/Service/ResourceService.php

```diff
@@ -138,4 +199,73 @@ public function getResourceById($id)
         );
         return $resource;
     }
+
+    /**
+     * Use a record driver to assign metadata to the current row.
+     *
+     * @param \VuFind\RecordDriver\AbstractBase $driver    The record driver
+     * @param \VuFind\Date\Converter            $converter Date converter
+     * @param \VuFind\Db\Entity\Resource        $resource  Resource entity
+     *
+     * @return \VuFind\Db\Entity\Resource
+     */
+    public function assignMetadata(
+        $driver,
+        \VuFind\Date\Converter $converter,
+        $resource
+    ) {
+        // Grab title -- we have to have something in this field!
+        $title = mb_substr(
```

Comment:
This is an existing problem, but we might as well fix it: it seems strange that we only do the substring on the primary title. It might be cleaner to:

```
$sortTitle = $driver->tryMethod('getSortTitle');
$title = mb_substr(
    !empty($sortTitle) ? $sortTitle : $driver->getBreadcrumb(),
    0,
    255,
    'UTF-8'
);
```